ffi/capi/src/bidi.rs

```diff
@@ -52,9 +52,11 @@ pub mod ffi {
         )]
         pub fn for_text<'text>(
             &self,
-            text: &'text str,
+            text: &'text DiplomatStr,
             default_level: u8,
         ) -> Box<ICU4XBidiInfo<'text>> {
+            #[allow(clippy::unwrap_used)] // #2520
+            let text = core::str::from_utf8(text).unwrap();
```

Comment:
This is the only user-triggerable panic left in `icu_capi`

Reply:
I think we should make this a 2.0 breaking thing to return an Option here since you're basically making the changes that fix most of https://github.com/unicode-org/icu4x/issues/2520

though I guess we can do that today without breaking ABI.

Reply:
Hmm, I don't want this to become fallible in other languages. I think we should reintroduce `str` here once we have Diplomat support https://github.com/rust-diplomat/diplomat/pull/369